pkg/common/cliprinter/cliprinter.go

```diff
@@ -0,0 +1,148 @@
+package cliprinter
+
+import (
+	"io"
+	"os"
+
+	"github.com/spiffe/spire/pkg/common/cliprinter/errorjson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/errorpretty"
+	"github.com/spiffe/spire/pkg/common/cliprinter/protojson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/protopretty"
+	"github.com/spiffe/spire/pkg/common/cliprinter/structjson"
+	"github.com/spiffe/spire/pkg/common/cliprinter/structpretty"
+	"google.golang.org/protobuf/proto"
+)
+
+type Printer interface {
+	MustPrintError(error)
+	MustPrintProto(...proto.Message)
+	MustPrintStruct(...interface{})
+
+	PrintError(error)
+	PrintProto(...proto.Message)
+	PrintStruct(...interface{})
+}
```

Comment:
I moved some stuff around following this comment and our conversation. Moving towards error return instead of bool cleaned some things up but I suspect maybe not as much as you'd hoped.

The implementations that return errors are now tucked inside an internal directory

Reply:
Awesome sauce.